src/digital_identity_service/index.js

```diff
@@ -201,6 +212,134 @@ class DigitalIdentityService {
       throw err;
     }
   }
+
+  /**
+   * @param {string} receiptId
+   */
+  async fetchReceiptById(receiptId) {
+    const receiptIdUrl = Buffer.from(receiptId, 'base64').toString('base64url');
+
+    const request = new RequestBuilder()
+      .withBaseUrl(this.apiUrl)
+      .withHeader('X-Yoti-Auth-Id', this.sdkId)
+      .withPemString(this.pem)
+      .withEndpoint(`/v2/receipts/${receiptIdUrl}`)
+      .withQueryParam('appId', this.sdkId)
+      .withMethod('GET')
+      .build();
+
+    try {
+      const response = await request.execute();
+      const parsedResponse = response.getParsedResponse();
+
+      return new ReceiptResponse(parsedResponse);
+    } catch (e) {
+      console.log(`fetching receipt error ${e}`);
+      throw e;
+    }
+  }
```

Comment:
This doesn't follow the pattern of the other methods, which have two `try` blocks. Otherwise, I don't see a problem with this approach. I'm actually not sure if the `try`s are necessary at all, since they just re-throw the error, and log something which I doubt will be useful to the user

Reply:
I agree. I think the approach from the other PRs has been to use the `DigitalIdentityServiceError` and throw that. It also contains some logic about networking errors. So should probably do the same here.